### DISCUSSION TITLE
Ajuda com javascript

### DISCUSSION THREAD

**Discussion by chastombergithub** [frontendbr/forum#2232] [Perguntas e Respostas]
Preciso retirar o random deste código e deixá-lo em ordem crescente para não embaralhar a frase. Segue abaixo o código.
`

<input id="entrada" type="text" onkeypress="embaralhar(event)" />

<label id="texto">

  <br/>

  <br/>

<button id="exibir" onClick="exibir()">Exibir Original

<button id="limpar" onClick="limpar()">Limpar</button>

var textoOriginal = '';

var entrada = document.getElementById("entrada");

function embaralhar(event) {

  var x = event.which || event.keyCode;

  if (x !== 8) { 

    textoOriginal += entrada.value.substr(entrada.value.length - 1);

    entrada.value = entrada.value.replaceAt(entrada.value.length -1, randomico());

  }

};

function exibir() {

  document.getElementById("texto").innerHTML = textoOriginal;

};

function limpar() {

  textoOriginal = '';

  document.getElementById("texto").innerHTML = '';

  document.getElementById("entrada").value = '';

};

function randomico() {

  var possible = "Preciso alterar por esta frase em ordem crescente";

  //var possible = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";

  return possible.charAt(Math.floor(Math.random() * possible.length));

}

String.prototype.replaceAt = function(index, character) {

  return this.substr(0, index) + character + this.substr(index + character.length);

}

**Comment by lipegusmao** [comment]
Já tentou colocar um sort() no lugar do Math.random()?

**Reply by chastombergithub** [reply]
tentei agora. não deu tb

**Reply by chastombergithub** [reply]
mas to vendo como alterar para este Sort que indicou. .obg

**Comment by caioagiani** [comment]
@chastombergithub tenta aqui: https://www.w3schools.com/jsref/jsref_sort.asp

**Reply by chastombergithub** [reply]
@caioagiani Obg. Tentei de algumas formas inseri sem sucesso.

sabe como eu deveria colocar? 

`function randomico() {
  //var possible = "ABCDE";
  //ANTIGO return possible.charAt(Math.floor(Math.random(0) * possible.length));
  var possible = ["A","B","C","D","E"];
  return possible.charAt(Math.floor(possible.sort(0) * possible.length));

}`

**Reply by caioagiani** [reply]
@chastombergithub tente:

```js
const sortAsc = (string) => string.split(" ").sort().join(" ");
const sortDesc = (string) => string.split(" ").reverse().join(" ");

const randomWord = "Aaa Bbb Eee Zzz Ccc Ddd Yyy";
console.log(sortAsc(randomWord)); // Aaa Bbb Ccc Ddd Eee Yyy Zzz
console.log(sortDesc(randomWord)); // Yyy Ddd Ccc Zzz Eee Bbb Aaa
```

**Reply by chastombergithub** [reply]
obg. Mas ele fez o mesmo que o código original colocando return sortAsc(randomWord);

A ideia do código original funciona pra mim exceto que ela esta em random.

O const randomWord = "Aaa Bbb Eee Zzz Ccc Ddd Yyy" apareceu no input ao dar o primeiro clique no teclado e não letra por letra

**Comment by vicentecrespop** [comment]
function randomico() {

var possible = "Preciso alterar por esta frase em ordem crescente";

//var possible = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
possible = possible.split('').sort()
while(possible[0] == ' '){
   possible.splice(0, 1)
}
return possible.join('')
}
